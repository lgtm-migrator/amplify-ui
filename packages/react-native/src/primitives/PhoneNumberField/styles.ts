import { StyleSheet } from 'react-native';

import { PhoneNumberFieldStyles } from './types';

export const styles: PhoneNumberFieldStyles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
});
