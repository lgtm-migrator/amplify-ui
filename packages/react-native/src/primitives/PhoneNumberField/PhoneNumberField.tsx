import React from 'react';

import { TextField } from '../TextField';
import { PhoneNumberFieldProps } from './types';
import { styles } from './styles';

// replace whitespace and add +
const phoneMask = (value: string | undefined) =>
  value ? `+${value.replaceAll(/\s/g, '')}` : '';

export default function PhoneNumberField({
  disabled,
  formatValue,
  fieldStyle,
  ...rest
}: PhoneNumberFieldProps): JSX.Element | null {
  return (
    <TextField
      {...rest}
      disabled={disabled}
      keyboardType="phone-pad"
      style={[styles.inputContainer, fieldStyle]}
      formatValue={formatValue ?? phoneMask}
    />
  );
}
