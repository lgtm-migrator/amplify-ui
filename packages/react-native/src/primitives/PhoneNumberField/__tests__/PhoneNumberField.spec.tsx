import React from 'react';
import { render } from '@testing-library/react-native';

import PhoneNumberField from '../PhoneNumberField';

const testID = 'phoneNumberInput';
const defaultProps = {
  testID,
};

// TODO: add more tests

describe('PhoneNumberField', () => {
  it('renders as expected', () => {
    const { toJSON, getByTestId } = render(
      <PhoneNumberField {...defaultProps} />
    );
    expect(toJSON()).toMatchSnapshot();

    const textInput = getByTestId(testID);
    expect(textInput.props.keyboardType).toBe('phone-pad');
    expect(textInput.props.editable).toBe(true);
    expect(textInput.props.accessible).toBe(true);
  });

  it('renders as expected when disabled', () => {
    const { toJSON, getByTestId } = render(
      <PhoneNumberField {...defaultProps} disabled />
    );
    expect(toJSON()).toMatchSnapshot();
    const textInput = getByTestId(testID);
    expect(textInput.props.editable).toBe(false);
  });

  it('renders as expected with formatValue', () => {
    const formatValue = (value: string | undefined) => {
      return `+${value}`;
    };
    const { toJSON, getByTestId } = render(
      <PhoneNumberField
        {...defaultProps}
        formatValue={formatValue}
        value="123"
      />
    );
    expect(toJSON()).toMatchSnapshot();
    const textInput = getByTestId(testID);
    expect(textInput.props.value).toBe('+123');
  });

  it('renders as expected without formatValue', () => {
    const { toJSON, getByTestId } = render(
      <PhoneNumberField {...defaultProps} value="1 2 3" />
    );
    expect(toJSON()).toMatchSnapshot();
    const textInput = getByTestId(testID);
    expect(textInput.props.value).toBe('+123');
  });
});
