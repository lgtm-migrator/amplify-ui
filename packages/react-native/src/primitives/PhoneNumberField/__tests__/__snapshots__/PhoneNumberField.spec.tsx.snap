// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumberField renders as expected 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": undefined,
    }
  }
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
      },
      Array [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <TextInput
      accessible={true}
      autoCapitalize="none"
      editable={true}
      keyboardType="phone-pad"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
      testID="phoneNumberInput"
      value=""
    />
  </View>
</View>
`;

exports[`PhoneNumberField renders as expected when disabled 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "opacity": 0.6,
      },
      Array [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <TextInput
      accessible={true}
      autoCapitalize="none"
      editable={false}
      keyboardType="phone-pad"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
      testID="phoneNumberInput"
      value=""
    />
  </View>
</View>
`;

exports[`PhoneNumberField renders as expected with formatValue 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": undefined,
    }
  }
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
      },
      Array [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <TextInput
      accessible={true}
      autoCapitalize="none"
      editable={true}
      keyboardType="phone-pad"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
      testID="phoneNumberInput"
      value="+123"
    />
  </View>
</View>
`;

exports[`PhoneNumberField renders as expected without formatValue 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": undefined,
    }
  }
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
      },
      Array [
        undefined,
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <TextInput
      accessible={true}
      autoCapitalize="none"
      editable={true}
      keyboardType="phone-pad"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
      testID="phoneNumberInput"
      value="+123"
    />
  </View>
</View>
`;
